front/components/assistant_builder/actions/configuration/ReasoningModelConfigurationSection.tsx

```diff
@@ -35,17 +91,25 @@ export function ReasoningModelConfigurationSection({
   });
   const { isDark } = useTheme();
 
+  // It should be selected by default, but in case it was not available,
+  // use the first one from the list (sorted by preference).
   const selectedReasoningModelConfig = useMemo(
     () =>
-      selectedReasoningModel &&
-      reasoningModels.find(
-        (m) =>
-          m.modelId === selectedReasoningModel.modelId &&
-          m.providerId === selectedReasoningModel.providerId
-      ),
+      (selectedReasoningModel &&
+        reasoningModels.find(
+          (m) =>
+            m.modelId === selectedReasoningModel.modelId &&
+            m.providerId === selectedReasoningModel.providerId
+        )) ??
+      reasoningModels[0],
```

Comment:
We already have the logic to show an error message in this component when no reasoning model is available https://github.com/dust-tt/dust/blob/717903b28becaabb195b6696902e2ef5def43c52/front/components/assistant_builder/actions/configuration/ReasoningModelConfigurationSection.tsx#L62-L73